### ISSUE TITLE
AppBar leading IconButton RTL bug in NEW VERSION (3.24)

### ISSUE THREAD

**Issue by ZhiaDev** [flutter/flutter#153621]
### Steps to reproduce

1. i updated flutter sdk to latest version (3.24.0)
2. run my project

### Expected results

Before I updated my Flutter, my Flutter version was 3.22.0, and this problem did not exist, and the icon was displayed correctly in its place, just like LTR, but after updating my Flutter to 3.24.0, the icon in RTL mode is outside the IconButton. It is displayed, but there is no problem in LTR mode

### Actual results

The icon inside IconButton is not displayed correctly in RTL mode, but in LTR mode

### Code sample

<details open><summary>Code sample</summary>

```dart
appBar: AppBar(
   title: Text('update'.tr),
   centerTitle: true,
   leading: IconButton(
   onPressed: () {
       Get.back();
        },
    icon: const Icon(Icons.arrow_back),
    ),
),
```

</details>

### Screenshots or Video

<details open>
<summary>Screenshots / Video demonstration</summary>

![rtl-problem](https://github.com/user-attachments/assets/40ae4990-133b-4354-b96b-9426920ed072)

</details>

### Logs

<details open><summary>Logs</summary>

```console
no log
```

</details>

### Flutter Doctor output

<details open><summary>Doctor output</summary>

```console
Doctor summary (to see all details, run flutter doctor -v):
[✓] Flutter (Channel stable, 3.24.0, on macOS 14.5 23F79 darwin-arm64, locale en)
[✓] Android toolchain - develop for Android devices (Android SDK version 34.0.0)
[✓] Xcode - develop for iOS and macOS (Xcode 15.4)
[✓] Chrome - develop for the web
[✓] Android Studio (version 2024.1)
[✓] VS Code (version 1.92.2)
[✓] Connected device (4 available)
[✓] Network resources

• No issues found!
```

</details>

**Comment by yazanturk** [comment]
@ZhiaDev  hello, for now chang flutter channel to master , its working for me

**Comment by jonahwilliams** [comment]
Duplicate of https://github.com/flutter/flutter/issues/153036

**Comment by mNouh94** [comment]
This issue has been fixed during update [#3.24.1](https://github.com/flutter/flutter/blob/stable/CHANGELOG.md#3241)

**Comment by ZhiaDev** [comment]
> This issue has been fixed during update [#3.24.1](https://github.com/flutter/flutter/blob/stable/CHANGELOG.md#3241)

Thank you Mohammed 🫶
